compute_tools/src/compute.rs

```diff
@@ -1222,21 +1224,24 @@ impl ComputeNode {
         let pgdata_path = Path::new(&self.pgdata);
         let postgresql_conf_path = pgdata_path.join("postgresql.conf");
         config::write_postgres_conf(&postgresql_conf_path, &spec, None)?;
-        // temporarily reset max_cluster_size in config
+
+        let compute_state = self.state.lock().unwrap().clone();
+        let max_concurrent_connections = self.max_service_connections(&compute_state, &spec);
+
+        // Temporarily reset max_cluster_size in config
         // to avoid the possibility of hitting the limit, while we are reconfiguring:
-        // creating new extensions, roles, etc...
+        // creating new extensions, roles, etc.
         config::with_compute_ctl_tmp_override(pgdata_path, "neon.max_cluster_size=-1", || {
             self.pg_reload_conf()?;
 
             if spec.mode == ComputeMode::Primary {
-                let mut url = self.connstr.clone();
-                url.query_pairs_mut()
-                    .append_pair("application_name", "apply_config");
-                let url = Arc::new(url);
+                let mut conf = tokio_postgres::Config::from_str(self.connstr.as_str()).unwrap();
+                conf.application_name("apply_config");
+                let conf = Arc::new(conf);
 
                 let spec = Arc::new(spec.clone());
 
-                self.apply_spec_sql(spec, url, 1)?;
```

Comment:
I think this `1` just meant that we still processed reconfiguration sequentially. And for projects with 1k DBs that'd timeout, compute will be recreated and stuck again. Refactored this to use the same logic. @MMeent thoughts?

Reply:
Reverted this back to 1, but here is the image that was built with higher concurrency. We can test it on the project with 10k DBs

Compute with higher apply config concurrency `neondatabase/vm-compute-node-v16:12071102747`
https://github.com/neondatabase/neon/actions/runs/12071102747/job/33663022735?pr=9919#step:8:1